src/classes/QueryPredicate.cls

```diff
@@ -0,0 +1,60 @@
+/*************************************************************************************************
+* This file is part of the Nebula Framework project, released under the MIT License.             *
+* See LICENSE file or go to https://github.com/jongpie/NebulaFramework for full license details. *
+*************************************************************************************************/
+public class QueryPredicate {
```

Comment:
It sounds pretty magical that you implemented a new object's classes in about a minute +1: